examples/model-score/app.py

```diff
@@ -0,0 +1,305 @@
+        input. This will be either the latest rows, or a specific time range. Also
+        filter out rows for models that the user has deselected.
+        """
+        data = recent_df() if input.timeframe() == "Latest" else timeframe_df()
+
+        # Filter the rows so we only include the desired models
+        return data[data["model"].isin(input.models())]
+
+    @reactive.Calc
+    def filtered_model_names():
+        return filtered_df()["model"].unique()
+
+    @output
+    @render.ui
+    def value_boxes():
+        data = filtered_df()
+        models = data["model"].unique().tolist()
+        scores_by_model = {
+            x: data[data["model"] == x].iloc[-1]["score"] for x in models
+        }
+        # Round scores to 2 decimal places
+        scores_by_model = {x: round(y, 2) for x, y in scores_by_model.items()}
+
+        return x.ui.layout_column_wrap(
+            "135px",
+            *[
+                # For each model, return a value_box with the score, colored based on
+                # how high the score is.
+                x.ui.value_box(
+                    model,
+                    ui.h2(score),
+                    theme_color="success"
+                    if score > THRESHOLD_MID
+                    else "warning"
+                    if score > THRESHOLD_LOW
+                    else "danger",
+                )
+                for model, score in scores_by_model.items()
+            ],
+            fixed_width=True,
+        )
+
+    @output
+    @render_plotly_streaming(recreate_key=filtered_model_names, update="data")
+    def plot_timeseries():
+        """
+        Returns a Plotly Figure visualization. Streams new data to the Plotly widget in
+        the browser whenever filtered_df() updates, and completely recreates the figure
+        when filtered_model_names() changes (see recreate_key=... above).
+        """
+        fig = px.line(
+            filtered_df(),
+            x="time",
+            y="score",
+            labels=dict(score="auc"),
+            color="model",
+            color_discrete_map=model_colors,
+            # The default for render_mode is "auto", which switches between
+            # type="scatter" and type="scattergl" depending on the number of data
+            # points. Switching that value breaks streaming updates, as the type
+            # property is read-only. Setting it to "webgl" keeps the type consistent.
+            render_mode="webgl",
+        )
+
+        fig.add_hline(
+            THRESHOLD_LOW,
+            line_dash="dash",
+            line=dict(color=THRESHOLD_LOW_COLOR, width=2),
+        )
+        fig.add_hline(
+            THRESHOLD_MID,
+            line_dash="dash",
+            line=dict(color=THRESHOLD_MID_COLOR, width=2),
+        )
+
+        fig.update_yaxes(range=[0, 1], fixedrange=True)
+        fig.update_xaxes(fixedrange=True, tickangle=60, dtick="M5")
```

Comment:
When switching to static view you get a very crowded x axis. Setting tick marks directly is a bit less noisy.

```suggestion
        fig.update_xaxes(fixedrange=True, tickangle=60, nticks=6)
```

Reply:
OK awesome, I wasn't sure how to do that.

Reply:
I got the best results by accepting the default.